### PR TITLE
Use v{} for version tags with pooch

### PR DESCRIPTION
**Pull request by jni**
## Description

The pooch URL was incorrect for released version tags given our convention to use v0.18.0 rather than just 0.18.0 for releases.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by jni** [comment]
Reference thread on Zulip: https://skimage.zulipchat.com/#narrow/stream/175598-general/topic/release.200.2E18/near/218161832

But, looks like this works for versions but not master... Hold please...

**Review by emmanuelle — APPROVED** [review]
Thanks @jni. pre failures have been addressed in #5103.

**Comment by emmanuelle** [comment]
We'll need a second person to review/approve/merge here.

File: skimage/data/__init__.py [hunk 1]
```diff
@@ -125,7 +125,12 @@ def create_image_fetcher():
     # This helps pooch understand that it should look in master
     # to find the required files
     pooch_version = __version__.replace('.dev', '+')
-    url = "https://github.com/scikit-image/scikit-image/raw/{version}/skimage/"
+    if '+' in pooch_version:
```
Comment (hmaarrfk) [inline skimage/data/__init__.py:128]: why is thel ogic different?

Comment (jni) [reply skimage/data/__init__.py:128]: @hmaarrfk for master, the data lives in `github.com/scikit-image/scikit-image/raw/master/skimage/...`, while for version tags, it's in `github.com/scikit-image/scikit-image/raw/v0.18.0/skimage/...`, for example.

Technically, the approach is not quite correct: the v0.18.x branch will still look in master. This happens to work, but might not always. For now, it gets us unstuck. In the future, maybe we move to setuptools_scm, which would solve a lot of issues more elegantly.

Comment (emmanuelle) [reply skimage/data/__init__.py:128]: This, and pooch knows how to convert internally `version` to `master` so that in the url it's `master`, I guess?

**Comment by emmanuelle** [comment]
@scikit-image/core please review/merge, this is a fix needed for the release.

**Review by mkcor — APPROVED** [review]

**Comment by emmanuelle** [comment]
Thank you all, merging.

**Comment by mkcor** [comment]
@emmanuelle I understand that the CI failure is resolved by #5109, right?

**Comment by emmanuelle** [comment]
@mkcor by #5103 (which itself is fixed by #5109)

**Comment by jni** [comment]
@meeseeksdev backport to v0.18.x

**Pull request merged**